### PR TITLE
Include new "channels" based endpoints to auth and cors flow + remove deprecated API endpoints. r=Yoric

### PR DESCRIPTION
**Pull request by azasypkin**

### CODE REVIEW

**Comment by azasypkin** [comment]
r? @Yoric 

I've removed deprecated methods assuming that you kept it just because of the client app - if not, tell me and I'll return it back.

**Comment by azasypkin** [comment]
> I've removed deprecated methods assuming that you kept it just because of the client app - if not, tell me and I'll return it back.

Also waiting for the tests to make sure they don't depend on deprecated API

File: src/http_server.rs [hunk 1]
```diff
@@ -87,12 +87,10 @@ impl<T: Controller> HttpServer<T> {
             // Taxonomy router paths. Keep in sync with taxonomy_router.rs
             (vec![Method::Get, Method::Post], "api/v1/services".to_owned()),
             (vec![Method::Post, Method::Delete], "api/v1/services/tags".to_owned()),
-            (vec![Method::Get, Method::Post], "api/v1/channels/getters".to_owned()),
-            (vec![Method::Get, Method::Post], "api/v1/channels/setters".to_owned()),
+            (vec![Method::Post], "api/v1/channels".to_owned()),
```
Comment (Yoric) [inline src/http_server.rs]: Why did you remove `Post`?

File: src/taxonomy_router.rs [hunk 2]
```diff
@@ -297,12 +262,10 @@ pub fn create<T>(controller: T, adapter_api: &Arc<AdapterManager>) -> Chain
         vec![
             AuthEndpoint(vec![Method::Get, Method::Post], "services".to_owned()),
             AuthEndpoint(vec![Method::Post, Method::Delete], "services/tags".to_owned()),
-            AuthEndpoint(vec![Method::Get, Method::Post], "channels/getters".to_owned()),
-            AuthEndpoint(vec![Method::Get, Method::Post], "channels/setters".to_owned()),
+            AuthEndpoint(vec![Method::Post], "channels".to_owned()),
```
Comment (Yoric) [inline src/taxonomy_router.rs]: Why did you remove `Get`?

File: src/taxonomy_router.rs [hunk 3]
```diff
@@ -297,12 +262,10 @@ pub fn create<T>(controller: T, adapter_api: &Arc<AdapterManager>) -> Chain
...
             AuthEndpoint(vec![Method::Get], "channels/get".to_owned()),
             AuthEndpoint(vec![Method::Put], "channels/set".to_owned()),
-            AuthEndpoint(vec![Method::Post, Method::Delete], "channels/getters/tags".to_owned()),
-            AuthEndpoint(vec![Method::Post, Method::Delete], "channels/setters/tags".to_owned())
+            AuthEndpoint(vec![Method::Post, Method::Delete], "channels/tags".to_owned())
```
Comment (Yoric) [inline src/taxonomy_router.rs:268]: Good catch, thanks.

File: src/http_server.rs [hunk 4]
```diff
@@ -87,12 +87,10 @@ impl<T: Controller> HttpServer<T> {
...
             (vec![Method::Put], "api/v1/channels/get".to_owned()),
             (vec![Method::Put], "api/v1/channels/set".to_owned()),
-            (vec![Method::Post, Method::Delete], "api/v1/channels/getters/tags".to_owned()),
-            (vec![Method::Post, Method::Delete], "api/v1/channels/setters/tags".to_owned())
+            (vec![Method::Post, Method::Delete], "api/v1/channels/tags".to_owned())
```
Comment (Yoric) [inline src/http_server.rs:93]: Good catch, thanks.

**Comment by Yoric** [comment]
Good idea. A few questions first, though.

File: src/http_server.rs (hunk 1 above)
Comment (azasypkin) [inline src/http_server.rs]: Ah you mean `GET`, right? I've removed `GET` and `POST` for old endpoints (`channels/getters` and `channels/setters`) and added `POST` for new endpoint (`channels`), but yeah, we probably still need `GET` as well...

File: src/taxonomy_router.rs (hunk 2 above)
Comment (azasypkin) [inline src/taxonomy_router.rs]: True, just missed it, will return it back, sorry

**Comment by azasypkin** [comment]
Hmm, funny :) Coverage decreased by 0.02% with  `+4 −43` changeset :)

@Yoric do you want me to write any test or it's okay to have 0.02% coverage decrease?

And I've handled your comments btw

**Comment by azasypkin** [comment]
Okay, now it's ready for the next round of review :) r? @Yoric

**Comment by Yoric** [comment]
r=me, thanks

**Comment by azasypkin** [comment]
Thanks for review, Travis is green, so merging.
